lib/item_repository.rb

```diff
@@ -5,62 +5,67 @@ class ItemRepository
   attr_reader :all
 
   def initialize(path)
-    @all = create_items(path)
+    @all = []
+    create_items(path)
   end
 
   def inspect
     "#<#{self.class} #{@items.size} rows>"
   end
 
   def create_items(path)
-    @all = []
     CSV.foreach(path, headers: true, header_converters: :symbol).each do |item|
       @all << Item.new(item, self)
     end
-    # item = CSV.load(path)
-    # item.map do |item_data|
-    #   Item.new(item_data)
-    # end
   end
 
   def find_by_id(id)
-    @all.find { |item| item.id == id }
+    @all.find do |item|
+      item.id == id
+    end
   end
 
   def find_by_name(name)
-    @all.find { |item| item.name.upcase == name.upcase }
+    @all.find do |item|
+      item.name.upcase == name.upcase
+    end
   end
 
   def find_all_with_description(description)
-    @all.find_all { |item| item.description == description }
+    @all.find_all do |item|
+      item.description == description
+    end
   end
 
   def find_all_by_price(price)
-    @all.find_all { |item| item.price == price }
+    @all.find_all do |item|
+      item.unit_price == price
+    end
   end
 
   def find_all_by_price_in_range(range)
-    @all.find_all { |item| range.include?(item.price) }
+    temp = @all.find_all do |item|
```

Comment:
Are you using this temp variable later? Don't see it within the method, so wondering if it's necessary.